src/Chapter2.hs

```diff
@@ -841,7 +872,12 @@ list.
 
 🕯 HINT: Use the 'cycle' function
 -}
-rotate = error "rotate: Not implemented!"
+
+rotate :: Int -> [a] -> [a]
+rotate i l
+	| i < 0 = []
+	| i == 0 = l
+	| otherwise = rotate (i - 1) (take (length l) $ tail (cycle l))
```

Comment:
Your implementation rotates a list by a single element N times. This is not so efficient. Taking elements from the end of the list and calculating list length is costly.